metadata-ingestion/src/datahub/ingestion/source/identity/azure_ad.py

```diff
@@ -205,6 +186,53 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
                     datahub_corp_user_urn_to_group_membership,
                 )
 
+    def _add_group_members_to_group_membership(
+        self,
+        parent_corp_group_urn: str,
+        azure_ad_group: dict,
+        user_urn_to_group_membership: Dict[str, GroupMembershipClass],
+    ) -> None:
+        # Extract and map members for each group
+        for azure_ad_group_members in self._get_azure_ad_group_members(azure_ad_group):
+            # if group doesn't have any members, continue
+            if not azure_ad_group_members:
+                continue
+            for azure_ad_member in azure_ad_group_members:
+                odata_type = azure_ad_member.get("@odata.type")
+                if odata_type == "#microsoft.graph.user":
+                    self._add_user_to_group_membership(
+                        azure_ad_member,
+                        parent_corp_group_urn,
+                        user_urn_to_group_membership,
+                    )
+                elif odata_type == "#microsoft.graph.group":
+                    # Since DataHub does not support nested group, we add the members to the parent group and not the nested one.
+                    self._add_group_members_to_group_membership(
+                        parent_corp_group_urn,
+                        azure_ad_member,
+                        user_urn_to_group_membership,
+                    )
+                else:
+                    raise ValueError(
+                        f"Unsupported @odata.type '{odata_type}' found in Azure group member"
+                    )
+
+    def _add_user_to_group_membership(
```

Comment:
Can we rearrange the parameters to this function to better match the `_add_group_members_to_group_membership` function of <group_urn>, <azure-object>, <map>

So just 

```
        self,
        group_urn: str,
        azure_ad_user: dict,
        user_urn_to_group_membership: Dict[str, GroupMembershipClass],
```